paymentsheet/src/main/java/com/stripe/android/paymentsheet/IntentConfirmationHandler.kt

```diff
@@ -198,13 +243,45 @@ internal class IntentConfirmationHandler(
             is InternalPaymentResult.Canceled -> Result.Canceled
         }
 
-        deferredIntentConfirmationType = null
-
-        completableResult?.complete(intentResult)
+        onIntentResult(intentResult)
 
         removeIsAwaitingForPaymentResult()
     }
 
+    private fun onExternalPaymentMethodResult(result: PaymentResult) {
+        val intentResult = currentArguments?.let { arguments ->
+            when (result) {
+                is PaymentResult.Completed -> Result.Succeeded(
+                    intent = arguments.intent,
+                    deferredIntentConfirmationType = null
+                )
+                is PaymentResult.Failed -> Result.Failed(
+                    cause = result.throwable,
+                    message = result.throwable.stripeErrorMessage(),
+                    type = ErrorType.ExternalPaymentMethod,
+                )
+                is PaymentResult.Canceled -> Result.Canceled
+            }
+        } ?: run {
+            val cause = IllegalStateException("Arguments should have been initialized before handling EPM result!")
+
+            Result.Failed(
+                cause = cause,
+                message = cause.stripeErrorMessage(),
+                type = ErrorType.ExternalPaymentMethod,
+            )
+        }
+
+        onIntentResult(intentResult)
+    }
+
+    private fun onIntentResult(result: Result) {
+        deferredIntentConfirmationType = null
+        currentArguments = null
+
+        completableResult?.complete(result)
```

Comment:
It would be a bug. I'm a little hesitant to add an error now because there were a few things that I noticed while moving stuff over in that we are not properly handling process death for Bacs & EPMs, which would likely cause this to trigger.

Reply:
I'm planning to do a quick follow-up to fix EPM process death handling